app/keepers/keepers.go

```diff
@@ -155,6 +156,10 @@ type AppKeepers struct {
 	keys    map[string]*storetypes.KVStoreKey
 	tkeys   map[string]*storetypes.TransientStoreKey
 	memKeys map[string]*storetypes.MemoryStoreKey
+
+	// encoding/decode helpers
+	Codec            codec.Codec
+	TxEncodingConfig client.TxEncodingConfig
```

Comment:
I needed this for the upgrade to parse the signed msgs, the option was to add it here or at the [CreateUpgradeHandler](https://github.com/babylonlabs-io/babylon/blob/5f8af8ced17d24f3f0c6172293cd37fb3d055807/app/upgrades/types.go#L29)